src/recorder.rs

```diff
@@ -79,6 +79,13 @@ impl Recorder {
                     }
                     Signal::Event(event) => {
                         self.events.push(event);
+
+                        // Record an entry early if we anticipate its serialized size will
+                        // be larger than 64kb. At the time of this writing, we assume each
+                        // event will be well under 256 bytes.
+                        if self.events.len() >= 65_536 / 256 {
```

Comment:
why not use size_of<Event>? or maybe static assert that size_of<Event> < 256?

Reply:
I'm fine with both these suggestions, but am not able to do it right now. If you take care of it for me, great. If not, I'll add that late tonight.